src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -175,16 +155,7 @@ class PrisonerIepLevelReviewService(
       throw NoDataFoundException(bookingId)
     }
 
-    prisonerIepLevelRepository.delete(prisonerIepLevel)
-    nextReviewDateUpdaterService.update(bookingId)
-
-    // If the deleted record had `current=true`, the latest IEP review becomes current
-    if (prisonerIepLevel.current) {
-      // The deleted record was current, set new current to the latest IEP review
-      prisonerIepLevelRepository.findFirstByBookingIdOrderByReviewTimeDesc(bookingId)?.run {
-        prisonerIepLevelRepository.save(this.copy(current = true))
-      }
-    }
+    incentiveStoreService.syncIncentiveRecords(prisonerIepLevel, bookingId)
```

Comment:
I don't mind extracting this logic into another service however I suggest the names are consistent to the existing method names to avoid confusion.

E.g. this method handle sync DELETE requests but the new method is called `syncIncentiveRecords ()`. By looking at this line alone I cannot really tell what this method does and I may well think it's updating multiple incentive records?